packages/firebase_database/lib/ui/firebase_list.dart

```diff
@@ -100,45 +95,57 @@ class FirebaseList extends ListBase<DataSnapshot>
   void _onChildAdded(Event event) {
     int index = 0;
     if (event.previousSiblingKey != null) {
-      index = _indexForKey(event.previousSiblingKey) + 1;
+      final _index = _indexForKey(event.previousSiblingKey!);
+      if (_index != null) {
```

Comment:
Those null checks are a behavior change

I'd use `!` instead of checking if `index` is null here.